### PR TITLE
[Entities/Repo/Controller] Employee + [Entities/Repo/Controller/Tests] FisaMedicala

### PR DESCRIPTION
**Pull request by OpariucRares**
#13 and #10 
To be done tests for employee

### CODE REVIEW

File: DucksNet.API.Integration Tests/BaseIntegrationTests.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.AspNetCore.Mvc.Testing;
+using SamuraiApp.Infrastructure.Sqlite;
```
Comment (Al-Andrew) [inline DucksNet.API.Integration Tests/BaseIntegrationTests.cs:2]: Aici presupun ca e typo de la cum o fost creat namespace-ul ala.

File: DucksNet.API.Integration Tests/DucksNet.API.Integration Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (Al-Andrew) [inline DucksNet.API.Integration Tests/DucksNet.API.Integration Tests.csproj:1]: Te rog fara spatii in nume. mai ales la chestii care pot fi path-uri

File: DucksNet.API.Integration Tests/Usings.cs [hunk 3]
```diff
@@ -0,0 +1,2 @@
+﻿global using Xunit;
+global using FluentAssertions;
```
Comment (Al-Andrew) [inline DucksNet.API.Integration Tests/Usings.cs:2]: What is this?

File: src/DucksNet.API/Controllers/EmployeesController.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+{
+    private readonly IRepository<Employee> repository;
+
+    public EmployeesController(IRepository<Employee> repository)
+    {
+        //to add the repository for sediu
+        this.repository = repository;
+    }
+    [HttpGet]
+    public IActionResult GetAll()
```
Comment (Al-Andrew) [inline src/DucksNet.API/Controllers/EmployeesController.cs:21]: No getByID?

File: src/DucksNet.API/Controllers/MedicalRecordsController.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    [HttpGet]
+    public IActionResult GetAll()
+    {
+        var cages = _medicalRecordRepository.GetAll();
+        return Ok(cages);
+    }
+    [HttpPost]
+    public IActionResult Create([FromBody] MedicalRecordDTO dto)
+    {
+        //var medicalRecord = MedicalRecord.Create(dto.IdAppointment, dto.IdClient);
```
Comment (Al-Andrew) [inline src/DucksNet.API/Controllers/MedicalRecordsController.cs:29]: Please don't leave code comments in finalized code.

File: src/DucksNet.API/DTO/EmployeeDTO.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿namespace DucksNet.API.DTO;
+
+public class EmployeeDTO
+{
+    public Guid IdSediu { get; set; }
+    public string Surname { get; set; }
```
Comment (Al-Andrew) [inline src/DucksNet.API/DTO/EmployeeDTO.cs:6]: Please fix build warnings. either by creating a constructor or by marking the fields as nullable.

File: src/DucksNet.Domain/Model/Employee.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
+﻿namespace DucksNet.Domain.Model;
+public class Employee
+{
+    public Employee(Guid idSediu, string surname, string firstName, string address, string ownerPhone, string ownerEmail)
```
Comment (Al-Andrew) [inline src/DucksNet.Domain/Model/Employee.cs:4]: Please keep the domain rich. Using Result with factory methods and private constructors. Also please don't pass Guids in the constructor, mark them nullable instead.

File: src/DucksNet.Domain/Model/MedicalRecord.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
+﻿using DucksNet.SharedKernel.Utils;
+
+namespace DucksNet.Domain.Model;
+public class MedicalRecord
+{
+
+    public MedicalRecord(Guid idAppointment, Guid idClient)
```
Comment (Al-Andrew) [inline src/DucksNet.Domain/Model/MedicalRecord.cs:7]: Same as with Employee. Please keep the domain rich.

File: src/DucksNet.Domain/Model/MedicalRecord.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        Id = Guid.NewGuid();
+        IdAppointment = idAppointment;
+        IdClient = idClient;
+    }
+
+    public Guid Id { get; private set; }
+    public Guid IdAppointment { get; private set; }
+    public Guid IdClient { get; private set; }
+
+    //TODO: to add other values (in the future) (RO)
```
Comment (Al-Andrew) [inline src/DucksNet.Domain/Model/MedicalRecord.cs:18]: Please follow the following convetion for TODOs/NOTEs/FIXMEs :
```c#
// TODO (<your name>): ...
```

File: src/DucksNet.Domain/Model/MedicalRecord.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    /*
+    public static Result<MedicalRecord> Create(Guid idAppointment, Guid idClient)
```
Comment (Al-Andrew) [inline src/DucksNet.Domain/Model/MedicalRecord.cs:20]: No code comments please.

File: src/DucksNet.Infrastructure/Sqlite/DatabaseContext.cs [hunk 11]
```diff
@@ -7,8 +7,15 @@ namespace SamuraiApp.Infrastructure.Sqlite;
 
 public class DatabaseContext : DbContext, IDatabaseContext
 {
-    DbSet<Cage> IDatabaseContext.Cages => Set<Cage>();
-    DbSet<CageTimeBlock> IDatabaseContext.CageTimeBlocks => Set<CageTimeBlock>();
+    public DatabaseContext()
+    {
+        this.Database.EnsureCreated(); //just for integration tests
```
Comment (Al-Andrew) [inline src/DucksNet.Infrastructure/Sqlite/DatabaseContext.cs:12]: Please make that comment a FIXME

File: src/DucksNet.Infrastructure/Sqlite/EmployeeRepository.cs [hunk 12]
```diff
@@ -0,0 +1,46 @@
+﻿using DucksNet.Domain.Model;
+using DucksNet.Infrastructure.Prelude;
+using DucksNet.SharedKernel.Utils;
+using SamuraiApp.Infrastructure.Sqlite;
+
+namespace DucksNet.Infrastructure.Sqlite;
+public class EmployeeRepository : IRepository<Employee>
+{
+    private readonly IDatabaseContext _employeeRepository;
```
Comment (Al-Andrew) [inline src/DucksNet.Infrastructure/Sqlite/EmployeeRepository.cs:9]: why is the `IDatabaseContext` called `_employeeRepository` here?

File: src/DucksNet.Infrastructure/Sqlite/MedicalRecordRepository.cs [hunk 13]
```diff
@@ -0,0 +1,46 @@
+﻿using DucksNet.Domain.Model;
+using DucksNet.Infrastructure.Prelude;
+using DucksNet.SharedKernel.Utils;
+using SamuraiApp.Infrastructure.Sqlite;
+
+namespace DucksNet.Infrastructure.Sqlite;
+public class MedicalRecordRepository : IRepository<MedicalRecord>
+{
+    private readonly IDatabaseContext databaseContext;
```
Comment (Al-Andrew) [inline src/DucksNet.Infrastructure/Sqlite/MedicalRecordRepository.cs:9]: Please maintain the naming convention here. `_databaseContext`

File: tests/DucksNet.UnitTests/MedicalRecordTests.cs [hunk 14]
```diff
@@ -0,0 +1,9 @@
+﻿namespace DucksNet.UnitTests;
+public class MedicalRecordTests
+{
+    [Fact]
+    public void When_CreateMedicalRecord_Then_ShouldReturnMedicalRecord()
+    {
+        //ToDo: i don't think it is needed here ?
```
Comment (Al-Andrew) [inline tests/DucksNet.UnitTests/MedicalRecordTests.cs:7]: Please maintain convetion for TODOs. Also please don't commit empty files unless the stubs are necessary for build to succeed.

File: tests/DucksNet.UnitTests/Using.cs [hunk 15]
```diff
@@ -0,0 +1,2 @@
+﻿global using Xunit;
+global using FluentAssertions;
```
Comment (Al-Andrew) [inline tests/DucksNet.UnitTests/Using.cs:2]: Again, what is this?

**Review by Al-Andrew — CHANGES_REQUESTED** [review]
Please also take a look over the 4 issues found by [codecover](https://www.codefactor.io/repository/github/ducks-net/sidiez/pull/28).

File: DucksNet.API.Integration Tests/Usings.cs (hunk 3 above)
Comment (OpariucRares) [reply DucksNet.API.Integration Tests/Usings.cs:2]: e generata de Visual Studio, ca sa nu scrii in fiecare unit test, scrii intr-un fisier si este global

Comment (Al-Andrew) [reply DucksNet.API.Integration Tests/Usings.cs:2]: aha... personal nu prea imi place dar e ok. :shipit:

**Review by Al-Andrew — APPROVED** [review]

**Pull request merged**